### PR TITLE
ci: Add markdownlint, test_html_build, and build_docs workflows

### PR DESCRIPTION
**Pull request by spetrosi**
- markdownlint runs against README.md to avoid any issues with
  converting it to HTML
- test_converting_readme converts README.md > HTML and uploads this test
  artifact to ensure that conversion works fine
- build_docs converts README.md > HTML and pushes the result to the
  docs branch to publish dosc to GitHub pages site.
- Fix markdown issues in README.md

Signed-off-by: Sergei Petrosian <spetrosi@redhat.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -275,7 +259,7 @@ Allowed values are `traditional`, `syslog`, and `modern`. Default to `modern`.
 
 **Note:** Unless the above options are given, these local file outputs are configured.
 
-```
+```bash
```
Comment (spetrosi) [inline README.md:262]: @richm markdownlint requires some syntax language to be specified with code blocks. Can you think of something other than bash in this case? The list of supported languages: https://docs.readme.com/rdmd/docs/code-blocks

File: README.md [hunk 2]
```diff
@@ -286,71 +270,74 @@ Allowed values are `traditional`, `syslog`, and `modern`. Default to `modern`.
...
                        `/path/to/output/dir/%FROMHOST%/%PROGRAMNAME:::secpath-replace%.log`
-- `remote_sub_path`: Relative path to logging_system_log_dir to store the filtered logs.
+* `remote_sub_path`: Relative path to logging_system_log_dir to store the filtered logs.
 
 **Note:** Selector options and property-based filter options are exclusive. If Property-based filter options are defined, selector options will be ignored.
 
 **Note:** If both `remote_log_path` and `remote_sub_path` are _not_ specified, the remote_file output configured with the following settings.
 
-```
+```java
```
Comment (spetrosi) [inline README.md:340]: And here I guess java is the one?

File: README.md [hunk 3]
```diff
@@ -52,25 +27,25 @@ satisfied.
 
 Otherwise, please run the following command line to install the collection.
 
-```
+```bash
```
Comment (spetrosi) [inline README.md:30]: @richm this Collection requirements section comes from template's README, right? We do not copy it automatically. It's identical in many roles so better to ask.

Comment (richm) [reply README.md:30]: > @richm this Collection requirements section comes from template's README, right?

Right.

> We do not copy it automatically.

Right.

> It's identical in many roles so better to ask.

If it is identical, that means it didn't need to be changed.

File: README.md (hunk 1 above)
Comment (richm) [reply README.md:262]: syntax language is required?  The original markdown spec says you can have fenced code blocks without specifying the syntax - https://www.markdownguide.org/extended-syntax/#fenced-code-blocks

File: README.md (hunk 2 above)
Comment (richm) [reply README.md:340]: Since it works - yes - but the rsyslog config language is not generally like java

File: README.md [hunk 4]
```diff
@@ -397,7 +384,7 @@ When TLS connection is configured, `ca_cert_src` and/or `ca_cert` is required.
 To configure the certificate of the logging system, `cert_src` and/or `cert` is required.
 To configure the private key of the logging system, `private_key_src` and/or `private_key` is required.
 
-```
+```yaml
```
Comment (richm) [inline README.md:387]: This needs to be reformatted entirely - see https://github.com/linux-system-roles/logging/pull/345

File: README.md [hunk 5]
```diff
@@ -906,16 +899,20 @@ associated TLS value.
...
-```
+
+```bash
 semanage port --list | grep "syslog"
 ```
+
 The newly specified port will be added to this default set.
-```
+
+```bash
```
Comment (richm) [inline README.md:915]: would be better to not have a syntax here - not sure why the processor does not allow code blocks without a syntax

**Pull request merged**